src/loader.js

```diff
@@ -146,15 +147,99 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    var iterator;
+    if (__global.Map && __global.Map.prototype && __global.Map.prototype.entries && __global.Map.prototype.keys
+         && __global.Map.prototype.values && __global.Map.prototype.get && __global.Map.prototype.set
+         && __global.Map.prototype.has && __global.Map.prototype.delete)
+      registry._registry.registryMap = new __global.Map();
+    else
+      registry._registry.registryMap = new InternalMapPolyfill();
+    // 4.4.2
+    if (__global.Symbol && __global.Symbol.iterator)
+      registry[__global.Symbol.iterator] = mapPolyfillEntriesIterator.bind(registry._registry.registryMap);
+    return registry;
+  }
+
+  // 4.2
+  function Registry() {
+    throw new Error('A registry may only be created when creating a loader');
+  }
```

Comment:
This may be a little too strict. This would only be necessary if the Registry constructor was public, which I don't believe it is. Since this is the case, simply using an internal `Registry` class is fine since it will only be instantiated within 4.1.1 surely?